static/js/tools/timeline_util.ts

```diff
@@ -62,38 +62,41 @@ function updateUrl(param: UrlParam) {
   }
   // update statsVar
   if ("statsVar" in param) {
```

Comment:
I feel use an object/interface to wrap url data is very needed now. Can barely follow the if and else here :)

Reply:
I moved some logics into helper functions.

Reply:
I moved some logics into helper functions, hope it's easier to follow now

Reply:
looks much better!